oemoflex/tools/plots.py

```diff
@@ -0,0 +1,144 @@
+    df.columns = data.columns.to_flat_index()
+    for i in df.columns:
+        if i[0] == bus_name and not (i[1] == demand_name):
+            df[i] = df[i] * -1
+            df.rename(columns={i: labels_dict[i]}, inplace=True)
+        else:
+            df.rename(columns={i: labels_dict[i]}, inplace=True)
+
+    return df
+
+def filter_timeseries(df, timestamp_col, start_date, end_date):
+    r"""
+    Filters a dataframe with a timeseries from a start date to an end date.
+
+    Parameters
+    ---------------
+    'df' : pandas.DataFrame
+        Dataframe with timeseries.
+    'timestamp_col' : string
+        Column name of the column which contains the timestamps.
+    'start_date' : string
+        String with the start date for filtering in the format 'YYYY-MM-DD hh:mm:ss'.
+    'end_date' : string
+        String with the end date for filtering in the format 'YYYY-MM-DD hh:mm:ss'.
+
+    Returns
+    ----------
+    'df' : pandas.DataFrame
+        Filtered dataframe.
+    """
+    mask = (df[timestamp_col] >= start_date) & (df[timestamp_col] <= end_date)
+    df = df.loc[mask]
+    df = df.copy()
+
+    return df
+
+
+def stackplot(ax, df, colors_dict, x, y_stack_pos, y_stack_neg):
+    # pos_stack and neg_stack determine the stack order
+    colors=[]
+    labels=[]
+    y_pos = []
+
+    for i in y_stack_pos:
+        labels.append(i)
+        colors.append(colors_dict[i])
+        y_pos.append(df[i])
+
+    x = df[x]
+
+    y_pos = np.vstack(y_pos)
+    ax.stackplot(x, y_pos, colors=colors, labels=labels)
+
+    colors = []
+    labels = []
+    y_neg = []
+
+    for i in y_stack_neg:
+        labels.append(i)
+        colors.append(colors_dict[i])
+        y_neg.append(df[i])
+
+    y_neg = np.vstack(y_neg)
+    ax.stackplot(x, y_neg, colors=colors, labels=labels)
+
+
+def lineplot(ax, df, colors_dict, x, y_line):
+    y = y_line
+    for i in y:
+        ax.plot(df[x], df[i], color=colors_dict[i], label=i)
+
+
+def plot_dispatch(ax, df, colors_dict, labels_dict, start_date, end_date, x_timestamp,
+                  y_stack_pos, y_stack_neg, y_line, bus_name, demand_name):
+    df = map_labels(df, labels_dict, bus_name=bus_name, demand_name=demand_name)
+    if not (start_date is None and end_date is None):
+        df = filter_timeseries(df, x_timestamp, start_date, end_date)
+
+    stackplot(ax, df, colors_dict=colors_dict, x=x_timestamp,
+              y_stack_pos=y_stack_pos, y_stack_neg=y_stack_neg)
+    lineplot(ax, df, colors_dict=colors_dict, x=x_timestamp, y_line=y_line)
+
+
+def eng_format(ax, df, unit, conv_number):
+    formatter0 = EngFormatter(unit=unit)
+    ax.yaxis.set_major_formatter(formatter0)
+    df[df.select_dtypes(include=['number']).columns] *= conv_number
+    return df
+
+
+# import data and yaml files
+input_data = r'C:\Users\meinm\Documents\Git\oemof-B3-Ergebnisdaten\03_postprocessed\simple_model\sequences\bus\BB-electricity.csv'
+data = pd.read_csv(input_data, header=[0,1,2,3], parse_dates=[0])
```

Comment:
```suggestion
data = pd.read_csv(input_data, header=[0,1,2], parse_dates=[0], index_col=0)
```

Reply:
Correct read-in of the columns and index.

Reply:
Thanks! 
Why do I don't need to skip row 3 then?